### PR TITLE
Add `weighted_crossentropy` for imbalanced classification problems

### PR DESCRIPTION
**Pull request by staticfloat**
This is useful if, for example, you want to train with 75% of your data being of class 0, and 25% of your data being of class 1.

### CODE REVIEW

File: src/layers/stateless.jl [hunk 1]
```diff
@@ -4,8 +4,16 @@ using NNlib: log_fast
...
 
-crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat) =
-  -sum(y .* log_fast.(ŷ)) / size(y, 2)
+function crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat)
+  return -sum(y .* log_fast.(ŷ)) / size(y, 2)
+end
+
+function weighted_crossentropy(ŷ::AbstractVecOrMat, y::AbstractVecOrMat, w::AbstractVecOrMat)
+  # Here we directly weight `y` so that this reduces to `-sum( y .* log(ŷ) .* w)`
+  return crossentropy(ŷ, y .* w)
```
Comment (vchuravy) [inline src/layers/stateless.jl]: It is probably worth it to inline `crossentropy` manually here so that broadcast fusion can remove that extra allocation.

Comment (staticfloat) [reply src/layers/stateless.jl]: Ah, didn't think of that.  Good call.

**Comment by MikeInnes** [comment]
Would it make sense to have this as a keyword argument? With `weight = 1` by default it wouldn't add any overhead.

**Comment by MikeInnes** [comment]
I went ahead and tried that.

Thanks @staticfloat!

**Pull request merged**